src/components/util/makeClient.ts

```diff
@@ -1,15 +1,34 @@
-import { APIClient } from '@unicsmcr/unics_social_api_client';
+import { APIClient, GatewayPacketType } from '@unicsmcr/unics_social_api_client';
 import API_HOST from './APIHost';
+import { setQueueStatus, QueueStatus } from '../../store/slices/AuthSlice';
+import store from '../../store';
 
 export default function makeClient() {
 	const token = localStorage.getItem('jwt');
 	const apiBase = `${API_HOST}/api/v1`;
 	const useWss = process.env.NODE_ENV === 'production';
-	return new APIClient({
+	const apiClient = new APIClient({
 		token: token ?? undefined,
 		apiBase,
 		useWss
 	});
+	apiClient.initGateway();
+	addQueueListeners(apiClient);
+	return apiClient;
+}
+
+function addQueueListeners(apiClient: APIClient) {
+	apiClient.gateway?.on(GatewayPacketType.JoinDiscoveryQueue, () => {
```

Comment:
It would be cleaner to do an if/else null check on gateway here, and if it's undefined to throw an error maybe. It means you don't have to use the conditional gateway.